2020-11-09  Cody Do  <codyd@lnxsrv06.seas.ucla.edu>

        * timespec.h (timespec_cmp): Changed the timespec_cmp function to use
conditionals rather than arithmetic for a comparison. This will ensure that integer overflow won't occur.
--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2020-11-09 18:21:11.647729000 -0800
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+   if (a.tv_sec > b.tv_sec) {
+    return 1;
+  }
+  else if (a.tv_sec < b.tv_sec) {
+    return -1;
+  }
+  else {
+    return 0;
+  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
